op-node/rollup/sequencing/sequencer.go

```diff
@@ -650,12 +662,28 @@ func (d *Sequencer) Stop(ctx context.Context) (hash common.Hash, err error) {
 	if err := d.l.LockCtx(ctx); err != nil {
 		return common.Hash{}, err
 	}
-	defer d.l.Unlock()
 
 	if !d.active.Load() {
+		d.l.Unlock()
 		return common.Hash{}, ErrSequencerAlreadyStopped
 	}
 
+	// ensure latestHead has been updated to the latest sealed/gossiped block before stopping the sequencer
+	for d.latestHead.Hash != d.latestSealed.Hash {
+		latestHeadSet := make(chan struct{})
+		d.latestHeadSet = latestHeadSet
+		d.l.Unlock()
```

Comment:
This function will be entered when there are multiple calls, but since we acquire the lock again before continuing, it should be fine. Would appreciate a sanity check from @ajsutton

Reply:
There is a possibility that:

1. We get a call to Stop.  We are actively sequencing so pass the `if !d.active.Load()` check above and start the process of stopping.
2. `latestHead != latestSealed` so we release the lock
3. We get a second call to Stop which also enters this loop
4. We get the `forkChoiceUpdate` even which makes `latestHead == latestSealed`
5. Call 1 reacquires the lock and stops the sequencer
6. Call 2 reacquires the lock and stops the sequencer again even though it was already stopped.

From op-node's perspective I think it's ok, but it does mean that the second call would normally have gotten `ErrSequencerAlreadyStopped` but now appears to succeed.  If we wanted to be strict about it we should check the sequencer is active again after re-acquiring the lock.

Reply:
Good call out, I added a second `active` check after the lock re-aquisition